cmd/vault-subpath-proxy/kv_update_transport.go

```diff
@@ -75,7 +85,68 @@ func (k *kvKeyValidator) RoundTrip(r *http.Request) (*http.Response, error) {
 	}
 
 	r.Body = io.NopCloser(bytes.NewBuffer(requestBodyBytes))
-	return k.upstream.RoundTrip(r)
+	response, err := k.upstream.RoundTrip(r)
+	if err != nil {
+		return response, err
+	}
+
+	if k.synchronousSecretSync {
+		k.syncSecret(body.Data)
+	} else {
+		go k.syncSecret(body.Data)
+	}
+	return response, nil
+}
+
+func (k *kvUpdateTransport) syncSecret(data map[string]string) {
+	if data[vault.SecretSyncTargetNamepaceKey] == "" || data[vault.SecretSyncTargetNameKey] == "" {
+		return
+	}
+	// This is part of a long-running server, so give a gracious timeout
+	// to prevent stuck goroutines
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	for cluster, client := range k.kubeClients {
+		if data[vault.SecretSyncTargetClusterKey] != "" && data[vault.SecretSyncTargetClusterKey] != cluster {
```

Comment:
Could we get a `func targetsCluster(data map[string]string, cluster string) bool` or some GoDoc on the `SecretSyncTargetClusterKey` - an empty one means all clusters? I imagine we're duplicating this logic more than once and I needed to re-read things to remember the semantics

Reply:
Added that to the vault api